### PR TITLE
DM-31261: Moved all python modules into the lsst.ts.dream.csc module …

### PR DESCRIPTION
**Pull request by wvreeven**
…in preparation for the new ts_dream_common package.

### CODE REVIEW

File: bin/run_dream.py [hunk 1]
```diff
@@ -24,12 +24,12 @@
 import asyncio
 import logging
 
-from lsst.ts import dream
+from lsst.ts.dream import csc
 
 logging.basicConfig(
```
Comment (tribeiro) [inline bin/run_dream.py:29]: Consider removing this setup here.

File: conda/meta.yaml [hunk 2]
```diff
@@ -24,6 +24,7 @@ test:
     - ts-idl {{ idl_version }}
     - ts-salobj {{ salobj_version }}
     - ts-tcpip
+    - ts-dream-common
```
Comment (tribeiro) [inline conda/meta.yaml:27]: you probably need to add `ts-dream-common` to the "run" requirements.. Also, I noticed `ts-tcpip` is not part of the run requirements.

File: doc/developer-guide/developer-guide.rst [hunk 3]
```diff
@@ -14,11 +14,11 @@ Dependencies
...
 
-* ts_xml - v9.0.0
+* ts_xml - v9.1
 * `SAL <https://ts-sal.lsst.io>`_ - v5.1.1
-* ts_salobj - v6.4.1
-* ts_idl - v3.1.3
-* ts_tcpip - v0.3.0
+* ts_salobj - v6.5
+* ts_idl - v3.2
+* ts_tcpip - v0.3
```
Comment (tribeiro) [inline doc/developer-guide/developer-guide.rst:21]: what about `ts-tcpip`?

File: python/lsst/ts/dream/csc/mock/mock_dream.py [hunk 4]
```diff
@@ -152,23 +156,23 @@ async def exit(self) -> None:
...
         """Close the hatch."""
         self.log.info("close called.")
 
-    async def stop(self):
+    async def stop(self) -> None:
         """Immediately stop operations and close the hatch."""
         self.log.info("stop called.")
 
-    async def ready_for_data(self, ready: bool):
+    async def ready_for_data(self, ready: bool) -> None:
```
Comment (tribeiro) [inline python/lsst/ts/dream/csc/mock/mock_dream.py:175]: This got renamed no?

File: python/lsst/ts/dream/csc/mock/mock_dream.py [hunk 5]
```diff
@@ -179,14 +183,14 @@ async def ready_for_data(self, ready: bool):
         """
         self.log.info(f"readyForData called with param ready {ready}")
 
-    async def data_archived(self):
+    async def data_archived(self) -> None:
```
Comment (tribeiro) [inline python/lsst/ts/dream/csc/mock/mock_dream.py:186]: This got renamed no?

**Review by tribeiro — APPROVED** [review]
Looks good! I added some small comments inline. Also, I think it would be a good idea to make alpha releases for the packages and make sure it all works as expected...

File: bin/run_dream.py (hunk 1 above)
Comment (wvreeven) [reply bin/run_dream.py:29]: Done

File: conda/meta.yaml (hunk 2 above)
Comment (wvreeven) [reply conda/meta.yaml:27]: Done

Comment (wvreeven) [reply conda/meta.yaml:27]: Done

File: doc/developer-guide/developer-guide.rst (hunk 3 above)
Comment (wvreeven) [reply doc/developer-guide/developer-guide.rst:21]: Do you mean ts_dream_common? Yeah, I'll add that.

File: python/lsst/ts/dream/csc/mock/mock_dream.py (hunk 4 above)
Comment (wvreeven) [reply python/lsst/ts/dream/csc/mock/mock_dream.py:175]: Yes, I was waiting for you to finish your review of `ts_dream_common`. Done now.

File: python/lsst/ts/dream/csc/mock/mock_dream.py (hunk 5 above)
Comment (wvreeven) [reply python/lsst/ts/dream/csc/mock/mock_dream.py:186]: Yes, I was waiting for you to finish your review of `ts_dream_common`. Done now.

**Pull request merged**